sbi_lens/simulator/Lpt_field.py

```diff
@@ -0,0 +1,243 @@
+import astropy.units as u
+import jax
+import jax.numpy as jnp
+import jax_cosmo as jc
+import jax_cosmo.constants as constants
+import numpy as np
+import numpyro
+import numpyro.distributions as dist
+from jax.scipy.ndimage import map_coordinates
+from jax_cosmo.scipy.integrate import simps
+from jaxpm.kernels import fftk
+from jaxpm.painting import cic_paint, compensate_cic
+from jaxpm.pm import growth_factor, pm_forces
+
+from sbi_lens.simulator.redshift import subdivide
+
+# code from François Lanusse
+
+
+def linear_field(mesh_shape, box_size, pk, field):
+    """
+    Generate initial conditions.
+    """
+    kvec = fftk(mesh_shape)
+    kmesh = (
+        sum((kk / box_size[i] * mesh_shape[i]) ** 2 for i, kk in enumerate(kvec)) ** 0.5
+    )
+    pkmesh = (
+        pk(kmesh)
+        * (mesh_shape[0] * mesh_shape[1] * mesh_shape[2])
+        / (box_size[0] * box_size[1] * box_size[2])
+    )
+
+    field = jnp.fft.rfftn(field) * pkmesh**0.5
+    field = jnp.fft.irfftn(field)
+    return field
+
+
+def lpt_lightcone(cosmo, initial_conditions, positions, a, mesh_shape):
+    """
+    Computes first order LPT displacement
+    """
+    initial_force = pm_forces(positions, delta=initial_conditions).reshape(
+        mesh_shape + [3]
+    )
+    a = jnp.atleast_1d(a)
+    dx = growth_factor(cosmo, a).reshape([1, 1, -1, 1]) * initial_force
+    return dx.reshape([-1, 3])
+
+
+def convergence_Born(cosmo, density_planes, r, a, dx, dz, coords, z_source):
```

Comment:
It's better to keep consistency in naming throughout the code.
Typically capital letter are only used for Classes and CONSTANTS.
functions => `snake_case` 
classes => `CamelCase`
```suggestion
def convergence_born(cosmo, density_planes, r, a, dx, dz, coords, z_source):
```